src/Compilers/CSharp/Portable/Operations/CSharpOperationFactory_Methods.cs

```diff
@@ -103,7 +103,14 @@ private IVariableDeclaratorOperation CreateVariableDeclaratorInternal(BoundLocal
             }
 
             // Static members cannot have an implicit this receiver
-            if (symbol != null && symbol.IsStatic && instance.WasCompilerGenerated && instance.Kind == BoundKind.ThisReference)
+            if (symbol?.IsStatic == true && instance.WasCompilerGenerated && instance.Kind == BoundKind.ThisReference)
+            {
+                return null;
+            }
+
+            // Local functions are not invoked on receivers from a language point of view, and thus we do not expose
+            // a receiver for them.
+            if (symbol is MethodSymbol { MethodKind: MethodKind.LocalFunction })
```

Comment:
>if (symbol is MethodSymbol { MethodKind: MethodKind.LocalFunction }) [](http://example.com/codeflow?start=12&length=68)

Would it be better to adjust binding so that it wouldn't add a receiver?

Reply:
I guess we are doing this for regular static methods too